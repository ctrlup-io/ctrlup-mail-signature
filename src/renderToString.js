import ReactDOM from "react-dom/server";
import { ServerStyleSheets } from "@material-ui/core/styles";

export default function renderToString(element) {
  const sheets = new ServerStyleSheets();
  const html = ReactDOM.renderToString(sheets.collect(element));
  const css = sheets.toString();
  return `
      <div>
        <style>${css}</style>
        <div>${html}</div>
      </div>
    `;
}
