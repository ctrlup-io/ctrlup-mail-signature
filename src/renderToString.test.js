import { styled } from "@material-ui/core/styles";
import React from "react";

import renderToString from "./renderToString";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useLayoutEffect: jest.requireActual("react").useEffect,
}));

const Button = styled("button")({
  background: "red",
});

describe("renderToString", () => {
  it("renders correctly", () => {
    expect(renderToString(<Button>Add</Button>)).toMatchSnapshot();
  });
});
