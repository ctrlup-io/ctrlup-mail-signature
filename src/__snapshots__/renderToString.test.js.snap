// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderToString renders correctly 1`] = `
"
      <div>
        <style>.button-root-1 {
  background: red;
}</style>
        <div><button class=\\"button-root-1\\">Add</button></div>
      </div>
    "
`;
